plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go

```diff
@@ -0,0 +1,137 @@
+	ClusterName     string `json:"cluster_name"     jsonschema:"description=The name of your EKS Cluster,required"`
+	UseAsync        bool   `json:"use_async"        jsonschema:"description=If true then async extraction optimization is enabled (Default: true)"`
+	PollingInterval uint64 `json:"polling_interval" jsonschema:"description=Polling Interval in seconds (default: 5s)"`
+	Shift           uint64 `json:"shift"            jsonschema:"description=Time shift in past in seconds (default: 1s)"`
+	BufferSize      uint64 `json:"buffer_size"      jsonschema:"description=Buffer Size (default: 200)"`
+}
+
+func (k *Plugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:          9,
+		Name:        pluginName,
+		Description: "Read Kubernetes Audit Events for EKS from Cloudwatch Logs",
+		Contact:     "github.com/falcosecurity/plugins",
+		Version:     "0.1.0",
+		EventSource: "k8s_audit",
+	}
+}
+
+// setDefault sets the configuration to its default values
+func (k *PluginConfig) setDefault() {
+	if p := os.Getenv("AWS_DEFAULT_PROFILE"); p != "" {
+		k.Profile = p
+	}
+	if p := os.Getenv("AWS_PROFILE"); p != "" {
+		k.Profile = p
+	}
+	if r := os.Getenv("AWS_DEFAULT_REGION"); r != "" {
+		k.Region = r
+	}
+	if r := os.Getenv("AWS_REGION"); r != "" {
+		k.Region = r
+	}
+	if k.Region == "" {
+		k.Region = "us-east-1"
+	}
+	k.UseAsync = true
+	// for PollingInterval, Shift and BufferSize, the default values from the package are used automatically
+}
+
+func (k *Plugin) Init(cfg string) error {
+	// read configuration
+	k.Config.setDefault()
+	err := json.Unmarshal([]byte(cfg), &k.Config)
+	if err != nil {
+		return err
+	}
+
+	// setup optional async extraction optimization
+	extract.SetAsync(k.Config.UseAsync)
+
+	return nil
+}
+
+func (p *Plugin) InitSchema() *sdk.SchemaInfo {
+	reflector := jsonschema.Reflector{
+		// all properties are optional by default
+		RequiredFromJSONSchemaTags: true,
+		// unrecognized properties don't cause a parsing failures
+		AllowAdditionalProperties: true,
+	}
+	if schema, err := reflector.Reflect(&PluginConfig{}).MarshalJSON(); err == nil {
+		return &sdk.SchemaInfo{
+			Schema: string(schema),
+		}
+	}
+	return nil
+}
+
+func (k *Plugin) Open(params string) (source.Instance, error) {
+	filter := cloudwatchlogs.CreateFilter("", "/aws/eks/"+k.Config.ClusterName+"/cluster", "kube-apiserver-audit", nil)
+	client := cloudwatchlogs.CreateClient(session.CreateSession(k.Config.Region, k.Config.Profile), nil)
+	ctx, cancel := context.WithCancel(context.Background())
+	options := cloudwatchlogs.CreateOptions(
+		time.Duration(k.Config.Shift),
+		time.Duration(k.Config.PollingInterval*uint64(time.Second)),
+		k.Config.BufferSize,
+	)
+	eventsC, errC := client.Open(ctx, filter, options)
+	pushEventC := make(chan source.PushEvent)
+	go func() {
+		for {
+			select {
+			case i := <-eventsC:
+				pushEventC <- source.PushEvent{Data: []byte(*i.Message)}
```

Comment:
You should use the `k8saudit.ParseAuditEventsPayload` method here before pushing to the channel. That would allow for content checking and timestamp setting. See how the `parseAuditEventsAndPush` private method is implemented (and how it logs errors).

Reply:
:+1: